specification/botservice/resource-manager/Microsoft.BotService/preview/2018-07-12/botservice.json

```diff
@@ -1012,6 +1012,283 @@
+              "$ref": "#/definitions/Error"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.BotService/enterpriseChannels/{resourceName}": {
+      "put": {
+        "tags": [
+          "Enterprise Channel"
+        ],
+        "description": "Creates an Enterprise Channel.",
+        "operationId": "EnterpriseChannels_Create",
+        "x-ms-examples": {
+          "Create Bot": {
+            "$ref": "./examples/CreateEnterpriseChannel.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            },
+            "description": "The parameters to provide for the new Enterprise Channel."
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "If resource is created successfully or already existed, the service should return 200 (OK).",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "201": {
+            "description": "If resource is created successfully, the service should return 201 (Created). Execution to continue asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Enterprise Channel"
+        ],
+        "description": "Updates an Enterprise Channel.",
+        "operationId": "EnterpriseChannels_Update",
+        "x-ms-examples": {
+          "Create Bot": {
+            "$ref": "./examples/UpdateEnterpriseChannel.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "x-ms-client-flatten": true,
+            "schema": {
+              "$ref": "#/definitions/EnterpriseChannel"
+            },
+            "description": "The parameters to provide to update the Enterprise Channel."
```

Comment:
This will need [`x-ms-parameter-location`](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-parameter-location).

Reply:
Add it to the body parameter? Docs mention that 
"This extension can only be applied on global parameters. If this is applied on any parameter in an operation then it will be ignored."

resourceGroupNameParameter and resourceNameParameter have x-ms-parameter-location as method already.